src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -50,8 +50,7 @@
       Text="RuntimeIdentifier is required for native compilation. Try running dotnet publish with the -r option value specified." />
     <Error Condition="'$(GeneratePackageOnBuild)' == 'true'" Text="GeneratePackageOnBuild is not supported for native compilation." />
     <Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != '' and '$(CustomNativeMain)' != 'true'" Text="NativeLib requires OutputType=Library." />
-    <!-- See https://github.com/dotnet/runtime/issues/89346 for details -->
-    <Error Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true'" Text="EventSource is not supported when compiling to a native library. Set EventSourceSupport to false." />
+    <Warning Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true' and '$(_SuppressNativeLibEventSourceWarning)' != 'true'" Text="Multiple components (including CoreCLR) with EventSource enabled can't be loaded into the same process: see https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/diagnostics for details and workarounds. EventSource should only be enabled in libraries if you are certain it will be the only component with EventSource loaded into the process. To suppress this warning you can set  the _SuppressNativeLibEventSourceWarning MSBuild property to true." />
```

Comment:
> Multiple components (including CoreCLR) with EventSource enabled can't be loaded into the same process

This is not a 100% correct statement. For example, .NET Framework (that has EventSource enabled by default) and native AOT compiled library with EventSource enabled can co-exist in the same process just fine.

What happens when you try to load two native AOT compiled libraries with EventSource enabled into the same process? Is it actually going to fail, or is the problem just that tracing won't work well for one of them?

Reply:
> What happens when you try to load two native AOT compiled libraries with EventSource enabled into the same process? Is it actually going to fail, or is the problem just that tracing won't work well for one of them?

We have an issue on it (#88087) that says the behavior is undefined.

Do you have a suggestion for a different wording? My suggestion is the wording that was there in the second to last commit.

Reply:
I think it is enough for the error message to say that this does not work and provide a link. Same as what we do e.g. for WinForms (`WinForms is not supported or recommended with trimming enabled. Please go to https:
//aka.ms/dotnet-illink/winforms for more details.`). It is compilated to describe the maze of what may or may not work in a succinct error message. You have to be a diagnostics expert to navigate that maze.

Based on the WPF and WinForms prior art, my suggested wording would be `EventSource is not supported or recommended when compiling to a native library. Please go to https://github.com/dotnet/runtime/issues/91762 for more details.`.

If we want to use a link that does not point to a dotnet/runtime issue, it should be point to a place that actually has more relevant details. https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/diagnostics does not have relevant details and I do not think we would want to add the relevant details to that page. (The WinForms and WPF links have the same problem - the page that it points to does not have relevant details for WinForms and WPF trimming.)

Reply:
> The WinForms and WPF links have the same problem - the page that they point to does not have relevant details for WinForms and WPF trimming.

@agocke @LakshanF - looks like you own those links. I'm guessing they should be:
- https://aka.ms/dotnet-illink/wpf -> https://learn.microsoft.com/dotnet/core/deploying/trimming/incompatibilities#wpf
- https://aka.ms/dotnet-illink/windows-forms -> https://learn.microsoft.com/dotnet/core/deploying/trimming/incompatibilities#windows-forms

Reply:
```suggestion
    <Warning Condition="'$(NativeLib)' != '' and '$(EventSourceSupport)' == 'true' and '$(_SuppressNativeLibEventSourceWarning)' != 'true'" Text="EventSource is not supported or recommended when compiling to a native library. Please go to https://github.com/dotnet/runtime/issues/91762 for more details." />
```